### DISCUSSION TITLE
Fertig gelötete Platinen / Bauteil-Kits

### DISCUSSION THREAD

**Discussion by biologist79** [biologist79/ESPuino#68] [Ideas]
Wollte auf diesem Wege mal anfragen, ob es eigentlich Interesse gibt, fertig gelötete Platinen oder Kits anzubieten. Nicht, dass ich sowas schon daheim hätte, aber vielleicht wäre das ja eine Option, sowas anzubieten.

Das Angebot richtet sich jetzt erstmal nur an den deutschsprachigen Raum, deswegen schreibe ich es hier auch in Deutsch rein.

**Comment by Soehm** [comment]
Hey,

nur damit ich das richtig verstehe, du sprichst von so etwas in der Art:

![ESP_customized](https://user-images.githubusercontent.com/77161403/105207446-d4725380-5b47-11eb-914d-323b90681c4c.jpg)
Größe: ca 6 x 7cm

Die Bauteile sind aus der Idee von @compactflash "übernommen". Den TDA1308 habe ich gegen einen LM4808 getauscht. Es fehlt "unten links" auf der Platine noch die Versorgungsmöglichkeit über Batterie/Akku. Zusätzlich müsste ich noch den "Headphone-Detect" von PIN22 zum MUTE im PAM8403 führen - ggfs. könnte auch ein Jumper entweder zu PIN22/ESP oder "MUTE" des PAM switchen.

Die Zeichnung ist noch unsauber - ist ein genereller Test, ob es so funktionieren könnte...

Grüße
Stefan

**Reply by biologist79** [reply]
Grundsätzlich generell, aber im Speziellen doch erstmal auf die Carrier-Boards bezogen, die es aktuell schon gibt. Ich selbst habe ja mit der Software schon immer genug zu tun, so dass ich selbst noch nicht dazu gekommen bin, sowas zu planen :-) Hast da auch eine Ladeelektronik für Batteriebetrieb drauf?
Aber cool, gefällt mir.

**Reply by biologist79** [reply]
Vergiss meine Frage... hast es ja geschrieben. Eieiei :-)

**Comment by Soehm** [comment]
Ah, die TS würde ich (für mich) vermutlich noch rauswerfen und gegen Jumper ersetzen, die Taster von 3Pin Header auf 2Pin Header verkleinern...

**Reply by biologist79** [reply]
Was mir noch einfällt... ich brainstorme einfach mal :-)
a) Ich würde ich über einen ESP32-WROVER nachdenken. Nachteil ist zwar, dass man zwei GPIOs verliert (16+17), aber Vorteil ist, dass man PSRAM hat (was auch der Grund ist, weswegen diese beiden GPIOs intern gebraucht werden). Speziell für Webstreams wird das für Caching verwendet und ist von Vorteil. Vielleicht werde ich den in Zukunft auch mehr einbinden, aber das weiß ich noch nicht; grundsätzlich bin ich eher bestrebt, die Features auch weiterhin für ohne PSRAM gleichermaßen zu haben. Aber zB bei Bluetooth könnte man mit PSRAM Vorteile haben (habe ich allerdings noch nicht getestet; ist nur eine Idee/Mutmaßung).
b) Wie hast du die SD-Karte angebunden? Über SD-MMC hat man erheblich mehr Datendurchsatz und spart aber einen GPIO. Man muss an der Stelle halt beachten, dass die GPIOs bei MMC nicht änderbar sind. Speziell im Hinblick darauf, dass GPIO2 verwendet wird, muss man drauf achten, an diesem GPIO keinen PullUp einzubauen, da man den ESP32 sonst nicht mehr flashen kann.
c) Kürzlich hinzugefügt habe ich einen Pull-Request, um den RFID-Reader PN5180 zu unterstützen. Vorteil an dem ist, dass er deutlich sensitiver ist, als der RC522. Nachteil ist, dass er etwas teurer ist (5eur pro Stück), nur in China erhältlich und man braucht mind. zwei Pins mehr.
d) Nochmal zum Batteriebetrieb: Da kann man sich bei Wemos Lolin sicherlich Inspiration holen. Finde das da gut gelöst. Bei der AiO-Platine vom original Tonuino-Projekt kann man ja offenbar zwischen LiFePO4 und LiPo auswählen. Ob man das wirklich braucht weiß ich nicht, aber LiFePo4 gelten wohl als weniger gefährlich als LiPo.

**Reply by Soehm** [reply]
Hey,

zu

a) gerade bei compactflash schon geschrieben: WRoverB ist bei jlcpcb nicht verfügbar, daher der WRoom

b) SPI, nach deiner Ausführung kann ich aber mit wenig Aufwand auf SD-MMC schwenken

c) ich hab noch etliche 522er rumfliegen, mehr als ich Boxen bauen will :-)

d) habe ich mich wirklich noch nicht eingelesen. Ich lese mich da mal die Tage ein.

VG
Stefan

**Comment by compactflash** [comment]
Hallo @Soehm 
Anbei mal mein schon getesteter Aufbau . Funktioniert , aber ich habe beim Einschalten für ca. 2 Sekunden ein paar Knackser . Ich habe aber noch keine  Stützkondensatoren , wie in der Schaltung, implementiert . Ich hoffe es ist dann weg .
**Edit: erledigt , neues Schaltbild hinzugefügt** . Es ist deutlich besser, aber bei Akkubetrieb immer noch etwas vorhanden . Leider habe ich keinen Oszillographen um das mal zu erforschen .
** 26.01.2021 Ausgetrickst! Das Knacken passiert nur an meinem im Schreibtisch eingebautem USB-Netzteil . Masseschleife ??? Am Mac ist nichts .

![Audio_AiO](https://user-images.githubusercontent.com/56302484/105323591-9702da00-5bca-11eb-8807-842b7bf8125a.jpeg)



 Als Endstufe habe ich provisorisch einen YX8002 benutzt . Mit dem PAM 8403 habe ich auch getestet . Es sieht alles vielversprechend aus . 


Als Laderegler dachte ich an bq24091 für LiPo . Dieser Chip hat den Vorteil das SystemLoad und Charge gleichzeitig gehen und der ansonsten obligatorische Mosfet und die Diode am Ausgang entfallen können . Das verringert außerdem den Standby-Strom . Favorit wäre aber ein BQ25170. Der kann 3 Akkutypen , ist recht neu , aber leider gibt es bei TI noch kein ausführliches Datenblatt . 

Ich werde WroverB nehmen .

Ich werde keinen CP2102 verwenden , flashen und monitoring nur mit FTDI-Adapter möglich . Man muss dann die Box zwar aufschrauben , aber wenn @biologist79 in der Zukunft die Schlagzahl mal verringert ist es auch nur noch seltener erforderlich.

Warum benutzt du für SD-Reader den 75LVC125 . Ich habe einen Reader ohne jegliche Treiber-Elektronik nur mit ein paar Pullups ( an GPIO2 entfernt) und es funktioniert einwandfrei .  Anbei mal Daten dazu .
[Test SD_MMC_Lolin ESP32_D32_Pro.txt](https://github.com/biologist79/Tonuino-ESP32-I2S/files/5844343/Test.SD_MMC_Lolin.ESP32_D32_Pro.txt)

Leider habe ich momentan kaum Zeit , bleibe aber dran

VG Willmar

**Reply by Soehm** [reply]
Hi,

würdest du den mute vom PAM nutzen oder software-seitig den vom ESP?

Ich bin mir bei deiner Zeichnung nicht ganz sicher: unterbrichst du die Signale Nf_r/Nf_l durch einstecken des Kopfhörers? Hmm...

Den WRoverB würde ich auch nehmen, meine Lötkunst ist aber eher ausbaufähig, daher greife ich (erstmal) auf den von jlcpcb vorrätigen WRoom zurück - daher auch der LM4808

Ich habe nur SD-Reader mit 75LVC125 zu Hause. Wenn ich SD-MMC verwende müssen dann nur MOSI und CLK per 10k (eventuell höher?) an GND angeschlossen werden? CS/SS entfällt ja...

VG
Stefan

**Comment by drtrigon** [comment]
@biologist79 : Sind die Kits irgendwo erhältlich? Oder ist die Idee eingeschlafen?

**Reply by biologist79** [reply]
Nein, das ist absolut "vital" :-)
Schreibe mir einfach über mein [Forum](https://forum.espuino.de/) eine PN.
